### ISSUE TITLE
The /post/guid route is needed

### ISSUE THREAD

**Issue by poppastring** [poppastring/dasblog-core#156]
The permanent link is supposed to have the following format:

> _/post/ba1af8eb-2cfb-491d-b51d-ff179d0d0f46_

This allows each post to also have a permanent URL even though the title of the post might change.

The Guids are in the xml file, in the content folder, for example … content\2018-02-24.dayentry.xml

**Comment by mikedamay** [comment]
@poppastring
Parmalinks
--------------
The following is strictly about my trying to understand clearly what the requirements are and nothing else..

I agree with your position above (https://github.com/poppastring/dasblog-core/issues/156#issue-354976131) that permalinks should comprise post/guid.

However, on the face of it, that appears to contradict your position when we were discussing logging (https://github.com/poppastring/dasblog-core/issues/81#issuecomment-405135911) .  “Similarly if someone changes the title of the post they have to take responsibility for a change in the permalink title (title and post name are inexorably connected).” together with the notion that we require a readable title in the browser address bar when a reader selects a post.

You may be drawing a distinction between permalink and permalink title that I am missing.

If my current understanding of the requirements were correct then we would not need website/post-title or website/date/post-title anywhere (and given previous discussions I can’t believe that is correct).

Please have a go at explaining and if we can’t do that in short order we may have to have a call.

I have an adjacent point on the use of root URL but one thing at a time, I think.

**Comment by poppastring** [comment]
@mikedamay There is one other consideration that I did not share at the time and that is related to _**RSS**_. 

RSS relies on each post having an ID that is not changeable and a URI that does not change as the syndication aggregators would have no way of determining that a post title we have changed is new or just modified.

An additional route that include "/guids/" are necessary to support RSS. I am not sure if we need to use this in any other context (there may be one or two cases but I am not sure what they are yet). The two related cases #157 and #158 need a way to describe valid permanent URI, not broken by title changes. The RSS feed also contains the title URI but that is handled slightly differently by RSS aggregators.


Here are a sample of the URIs RSS expects, including the normal URI:
```
<wfw:comment>https://www.poppastring.com/blog/CommentView,guid,bc7a29c7-767e-43c8-a03c-2a9ffa039b73.aspx</wfw:comment>
<wfw:commentRss>https://www.poppastring.com/blog/SyndicationService.asmx/GetEntryCommentsRss?guid=bc7a29c7-767e-43c8-a03c-2a9ffa039b73</wfw:commentRss>
<title>Should we use SecureString in ASP.NET for handling passwords?</title>
<guid isPermaLink="false">https://www.poppastring.com/blog/PermaLink,guid,bc7a29c7-767e-43c8-a03c-2a9ffa039b73.aspx</guid>
<link>https://www.poppastring.com/blog/ShouldWeUseSecureStringInASPNETForHandlingPasswords.aspx</link>
```

I hope this makes some sense, we can talk directly about this whole topic if I am missing something.

Thanks!
